aptos-move/framework/aptos-framework/sources/multisig_account.move

```diff
@@ -0,0 +1,1081 @@
+            emit_event(
+                &mut multisig_account_resource.approve_transaction_events,
+                ApproveTransactionEvent {
+                    owner: owner_addr,
+                    transaction_id,
+                    num_approvals: simple_map::length(&transaction.approvals),
+                }
+            );
+        } else {
+            simple_map::add(&mut transaction.rejections, address_of(owner), true);
+            emit_event(
+                &mut multisig_account_resource.reject_transaction_events,
+                RejectTransactionEvent {
+                    owner: owner_addr,
+                    transaction_id,
+                    num_rejections: simple_map::length(&transaction.rejections),
+                }
+            );
+        };
+    }
+
+    /// Remove the next transaction if it has sufficient owner rejections.
+    public entry fun remove_transaction(
+        owner: &signer,
+        multisig_account: address,
+    ) acquires MultisigAccount {
+        assert_multisig_account_exists(multisig_account);
+        let multisig_account_resource = borrow_global_mut<MultisigAccount>(multisig_account);
+        assert_is_owner(owner, multisig_account_resource);
+        let transaction_id = multisig_account_resource.last_transaction_id + 1;
+        assert!(
+            table::contains(&multisig_account_resource.transactions, transaction_id),
+            error::not_found(ETRANSACTION_NOT_FOUND),
+        );
+        let transaction = table::remove(&mut multisig_account_resource.transactions, transaction_id);
+        assert!(
+            simple_map::length(&transaction.rejections) >= multisig_account_resource.signatures_required,
+            error::invalid_state(ENOT_ENOUGH_REJECTIONS),
+        );
+
+        multisig_account_resource.last_transaction_id = transaction_id;
+        emit_event(
+            &mut multisig_account_resource.remove_transaction_events,
+            RemoveTransactionEvent {
+                transaction_id,
+                num_rejections: simple_map::length(&transaction.rejections),
+                executor: address_of(owner),
+            }
+        );
+    }
+
+    /// Execute the next transaction if it has enought approvals. This doesn't actually invoke the target function but
+    /// simply marks the transaction as already been executed (by removing it from the transactions table). Actual
+    /// function invocation is done as part executing the MultisigTransaction.
+    ///
+    /// This function is private so no other code can call this beside the VM itself as part of MultisigTransaction.
+    ///
+    /// @param target_function Optional and can be empty if the full transaction payload is stored on chain.
+    /// @param args Optional and can be empty if the full transaction payload is stored on chain.
+    /// @return The transaction payload to execute as the multisig account.
+    fun execute_transaction(
```

Comment:
Assume there is a multi-sign account with two owners[A,B] and a threshold of 2. 
If A creates two transactions and gets enough approves, but tx-1 changes the owner to [C,D], should tx-2 be executed?
If tx-2 not get enough approves, but tx-1 reduce the threshold, how to check if tx-2 is executable?

Reply:
This is a known constraint here. We could add some validations of existing approvals when counting them. But this is an extra step that I'm still considering whether is worth the effort if there's always some small trust assumption of owners in a multisig wallet.

Reply:
How about add a version number into the wallet and transaction struct?
1. when create a new transaction, set `tx.version = wallet.version`
2. when the wallet update, increase the wallet.version: `wallet.version =  wallet.version + 1`
3. the tx is executable only if  `tx.version == wallet.version`

Reply:
Or create a snapshot of `owners` and `signatures_required` for every transaction. The new owner does not affect the old transaction.

Reply:
Updated to go through the votes (approvals/rejections) when validating whether a multisig transaction is executable (multisig_account::validate_multisig_transaction). This means that if an owner was removed and previously approved some transactions, their approval would not count. I also added a test for this (test_validate_transaction_should_not_consider_removed_owners)